kernel/kernel-api/src/test/scala/io/delta/kernel/internal/checksum/ChecksumWriterSuite.scala

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.checksum
+
+import io.delta.kernel.data.Row
+import io.delta.kernel.internal.actions.{Format, Metadata, Protocol}
+import io.delta.kernel.internal.checksum.CRCInfo.CRC_FILE_SCHEMA
+import io.delta.kernel.internal.data.GenericRow
+import io.delta.kernel.internal.fs.Path
+import io.delta.kernel.internal.util.VectorUtils
+import io.delta.kernel.internal.util.VectorUtils.{stringArrayValue, stringStringMapValue}
+import io.delta.kernel.test.{BaseMockJsonHandler, MockEngineUtils}
+import io.delta.kernel.types.StructType
+import io.delta.kernel.utils.CloseableIterator
+import org.scalatest.funsuite.AnyFunSuite
+
+import java.util
+import java.util.{Collections, Optional}
+
+/**
+ * Test suite for ChecksumWriter functionality.
+ */
+class ChecksumWriterSuite extends AnyFunSuite with MockEngineUtils {
+
+  private val FAKE_DELTA_LOG_PATH = new Path("/path/to/delta/log")
+
+  // Schema field indices in crc file
+  private val TABLE_SIZE_BYTES = CRC_FILE_SCHEMA.indexOf("tableSizeBytes")
+  private val NUM_FILES = CRC_FILE_SCHEMA.indexOf("numFiles")
+  private val NUM_METADATA = CRC_FILE_SCHEMA.indexOf("numMetadata")
+  private val NUM_PROTOCOL = CRC_FILE_SCHEMA.indexOf("numProtocol")
+  private val TXN_ID = CRC_FILE_SCHEMA.indexOf("txnId")
+  private val METADATA = CRC_FILE_SCHEMA.indexOf("metadata")
+  private val PROTOCOL = CRC_FILE_SCHEMA.indexOf("protocol")
```

Comment:
if we only use these variables once I'd prefer to just put them in-line it's more self-defining. if not please add something like `_IDX` to them, it's not intuitive that they are indices